src/lib/autocomplete/OVERVIEW.md

```diff
@@ -0,0 +1,149 @@
+
+We already have access to the built-in `valueChanges` observable on the `FormControl`, so we can simply map the text 
+input's values to the suggested options by passing them through our filter function. The resulting observable 
+(`filteredOptions`) can be added to the template in place of the `options` property using the `async` pipe.
+
+Below we are also priming our value change stream with `null` so that the options are filtered by that value on init 
+(before there are any value changes).
+
+*my-comp.ts*
+```ts
+class MyComp {
+   myControl = new FormControl();
+   options = [
+    'One',
+    'Two',
+    'Three'
+   ];
+   filteredOptions: Observable<string[]>;
+
+   ngOnInit() {
+      this.filteredOptions = this.myControl.valueChanges
+         .startWith(null)
+         .map(val => val ? this.filter(val) : this.options.slice());
+   }
+   
+   filter(val: string): string[] {
+      return this.options.filter(option => option.match(new RegExp(val, 'gi'))); 
+   }
+}
+```
+
+*my-comp.html*
+```html
+<md-input-container>
+   <input type="text" mdInput [formControl]="myControl" [mdAutocomplete]="auto">
+</md-input-container>
+
+<md-autocomplete #auto="mdAutocomplete">
+   <md-option *ngFor="let option of filteredOptions | async" [value]="option">
+      {{ option }}
+   </md-option>
+</md-autocomplete>
+```
+
+### Setting separate control and display values
+
+If you want the option's control value (what is saved in the form) to be different than the option's display value 
+(what is displayed in the actual text field), you'll need to set the `displayWith` property on your autocomplete 
+element. A common use case for this might be if you want to save your data as an object, but display just one of 
+the option's string properties.
+
+To make this work, create a function on your component class that maps the control value to the desired display value. 
+Then bind it to the autocomplete's `displayWith` property. You'll probably also want to add a step to your 
+`valueChanges` stream that maps the value through your display function. 
+
+```html
+<md-input-container>
+   <input type="text" mdInput [formControl]="myControl" [mdAutocomplete]="auto">
+</md-input-container>
+
+<md-autocomplete #auto="mdAutocomplete" [displayWith]="displayFn">
+   <md-option *ngFor="let option of filteredOptions | async" [value]="option">
+      {{ option }}
+   </md-option>
+</md-autocomplete>
+```
+
+*my-comp.ts*
+```ts
+class MyComp {
+   myControl = new FormControl();
+   options = [
+    {id: 1, name: 'One'},
+    {id: 2, name: 'Two'},
+    {id: 3, name: 'Three'}
+   ];
+   filteredOptions: Observable<any[]>;
+
+   ngOnInit() { 
+      this.filteredOptions = this.myControl.valueChanges
+         .startWith(null)
+         .map(val => this.displayFn(val))
+         .map(name => name ? this.filter(name) : this.options.slice());
+   }
+   
+   filter(val: string): any[] {
+      return this.options.filter(option => option.match(new RegExp(val, 'gi'))); 
```

Comment:
You typically don't want to use `match` when you are looking for a yes/no result. 
http://stackoverflow.com/a/10940138/477102